src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PreReleaseUserService.cs

```diff
@@ -260,7 +268,8 @@ await validEmails
                                     release,
                                     invite.Email.ToLower(),
                                     isNewUser: user == null)
-                                .OnSuccessDo(async () => await MarkInviteEmailAsSent(invite));
+                                .OnSuccessDo(async () =>
+                                    await _userReleaseInviteRepository.MarkEmailAsSent(invite));
```

Comment:
I believe I pulled this out expecting that I would be using it in ReleaseInviteService, but that ended up not being the case. I'm inclined to leave it now it's done, but can move it back if you'd prefer.

Reply:
In general I think its overly specific and I would keep the repository cleaner by limiting it to CRUD operations on the resource that are providing some value and reusable abstraction over the equivalent of what has to be done with the database context alone.

In this case 'MarkEmailAsSent' sounds like business logic that should be in a service rather than a repository and it provides nothing that can't be achieved by updating the invite and calling SaveChanges on it. If we were to write repository methods for updating resources for every place we allow updating a single field then they would be huge.

Reply:
I've moved it back